crates/bevy_pbr/src/light.rs

```diff
@@ -473,27 +477,25 @@ pub fn update_directional_light_cascades(
     }
 }
 
-fn calculate_cascade(
-    aspect_ratio: f32,
-    tan_half_fov: f32,
-    cascade_texture_size: f32,
-    light_to_world: Mat4,
-    camera_to_light: Mat4,
-    z_near: f32,
-    z_far: f32,
-) -> Cascade {
-    debug_assert!(z_near <= 0.0, "z_near {z_near} must be <= 0.0");
-    debug_assert!(z_far <= 0.0, "z_far {z_far} must be <= 0.0");
-    // NOTE: This whole function is very sensitive to floating point precision and instability and
-    // has followed instructions to avoid view dependence from the section on cascade shadow maps in
-    // Eric Lengyel's Foundations of Game Engine Development 2: Rendering. Be very careful when
-    // modifying this code!
+fn frustum_corners_ortho(area: Rect, z_near: f32, z_far: f32) -> [Vec3A; 8] {
+    // NOTE: These vertices are in the specific order required by [`calculate_cascade`].
+    [
+        Vec3A::new(area.max.x, area.min.y, z_near),
+        Vec3A::new(area.max.x, area.max.y, z_near),
+        Vec3A::new(area.min.x, area.max.y, z_near),
+        Vec3A::new(area.min.x, area.min.y, z_near),
+        Vec3A::new(area.min.x, area.min.y, z_far),
+        Vec3A::new(area.max.x, area.min.y, z_far),
```

Comment:
```suggestion
        Vec3A::new(area.max.x, area.max.y, z_far),
```
Top right should be at (max_x, max_y)